thorlcr/graph/thgraph.cpp

```diff
@@ -2757,6 +2757,8 @@ void CJobBase::applyMemorySettings(const char *context)
         unsigned totalRequirementsMB = (unsigned)(memorySpecifications["total"] / 0x100000);
         unsigned recommendedMaxMB = (unsigned)(memorySpecifications["recommendedMaxMemory"] / 0x100000);
         queryMemoryMB = recommendedMaxMB - totalRequirementsMB;
+        unsigned sharedInstances = globals->getPropInt(VStringBuffer("%s/@sharedInstances", memoryContext.str()), 1);
+        queryMemoryMB /= sharedInstances;
```

Comment:
Danger of div0?

Reply:
this is ok, it will either be absent (and default to 1 on previous line), or it has been checked/set to >0 and valid prior to this (by the code in thmastermain.cpp)